src/index.ts

```diff
@@ -13,23 +13,20 @@ async function run() {
     const architecture = process.arch;
     const workspaceDir = process.cwd();
 
+    // Export env variable to inject the buildtool plugin
+    core.exportVariable('MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE', 'matlab.ciplugins.github.getDefaultPlugins');
```

Comment:
It would be nice if this variable only existed during the run of MATLAB. `ExecOptions` has a [`env` property](https://github.com/actions/toolkit/blob/ae38557bb0dba824cdda26ce787bd6b66cf07a83/packages/exec/src/interfaces.ts#L10) you can use so that the environment variable only exists during the execution of the command, rather than changing the environment of the entire system.

Reply:
I think i would have to update `runCommand()` signature inorder to use `ExecOptions`  hence keeping it as is. 

```
const exec = require('@actions/exec');

/**
 * Executes a given command using a provided exec function.
 * 
 * @param {Function} execFunction - The exec function to use for command execution.
 * @param {string} command - The command to execute.
 * @param {string[]} args - Arguments for the command.
 * @param {exec.ExecOptions} [options] - Optional execution options.
 */
async function executeCommand(execFunction, command, args, options) {
  try {
    await execFunction(command, args, options);
    console.log(`Successfully executed command: ${command}`);
  } catch (error) {
    console.error(`An error occurred while executing command: ${command}`, error);
  }
}

// Example usage
const command = 'echo';
const args = ['Hello, world!'];
// Define custom environment variables via ExecOptions
const options = {
  env: {
    ...process.env,
    CUSTOM_VAR: 'MyCustomValue',
  },
};

executeCommand(exec.exec, command, args, options)
```

Reply:
I don't think you need to modify the `runCommand` signature. The `ExecFn` parameter can just be modified to pass in a function that uses your desired `ExecOptions`.

Reply:
One option is i could add options parameter https://github.com/matlab-actions/run-command/blob/79bb74bc15ad0a5cc38fccacdef7105153361f46/src/matlab.ts#L21 

My concern is can I achieve it without having to change anything in run-command ?

Reply:
You don't need to change anything in `run-command`. You can just use an anonymous function for the `ExecFn` argument:

```javascript
const execOptions = {env:{"MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE":"ciplugins.github.getDefaultPlugins"}};
await matlab.runCommand(helperScript, platform, architecture, (cmd,args)=>exec.exec(cmd,args,execOptions), startupOptions);
```

Reply:
Thanks Mark , i had to make some changes to include the process envvars as it could not find matlab with approach above.